src/packages/@ncigdc/components/analysis/SelectSet.js

```diff
@@ -84,14 +86,20 @@ const SetTable = ({
                 value={setId}
                 disabled={msg}
                 onChange={e => {
-                  const ids = xor(selectedSets[type], [e.target.value]);
-                  const sets = ids.length
-                    ? {
-                        ...selectedSets,
-                        [type]: ids,
-                      }
-                    : omit(selectedSets, type);
-                  setSelectedSets(sets);
+                  const typeSets = selectedSets[type] || {};
+                  const setId = e.target.value;
+                  const newTypeSet = typeSets[setId]
```

Comment:
took me a minute to understand what the change was here, but I see it now. Do you think you could add a comment explaining what 'typeSet' is in this context?

Reply:
Took me a second to understand too, I'll take a stab at making it clearer 👍

Reply:
Refactored so the on change looks like 
```
onChange={e => {
  const setId = e.target.value;
  const setIdPath = [type, setId];
  setSelectedSets(
    _.get(selectedSets, setIdPath)
      ? removeEmptyKeys(_.omit(selectedSets, setIdPath))
      : _.set({ ...selectedSets }, setIdPath, sets[setId]),
  );
}}
```
A bit clearer, what do you think?

Reply:
definitely clearer 👍